### ISSUE TITLE
Strange filenames returned on Amazon EC2 - 'PixelsPerCentimeter'

### ISSUE THREAD

**Issue by chrisfinch** [hacksparrow/node-easyimage#32]
Hi,
I am resizing a bunch of images on an EC2 instance and the names being returned are very strange and are preventing me from referencing the resized images. This all works fine on my local version.

Example resize call:

```
easyimg.resize({
  src: path,
  dst:__dirname + '/../public/uploads/460x276-2x' + '.png',
  width: 920,
  height: 552
})
```

Example returned data in promise:

```
{ type: 'PNG',
  depth: 8,
  width: 496,
  height: 552,
  size: 312674,
  density: 118.11,
  name: 'PixelsPerCentimeter 460x276-2x.png' }
```

As you can see the string `PixelsPerCentimeter ` is being prepended to the name attribute (the files appear to be named correctly in the filesystem...)

The EC2 instance is running on Ubuntu and I have ImageMagick installed as follows: `Version: ImageMagick 6.7.8-9 2014-05-12 Q16`

I'm pretty confused at this point having googled a lot for a solution.

Thanks in advance for any help.

**Comment by gzurbach** [comment]
I'm having the exact same issue. I tried to rename the image it doesn't solve the problem so it's not related to the name I think. I'd be happy to send you the image if you are interested.

I'm using easyimage in production and I had about a dozen images renamed to `Undefined the-original-name` after calling `resize` (out of 500 images uploaded). This is the first time I personally run into this bug but users already complained about it before.

Let me know if you are interested by the image, I'd need an email address where I can send it.

**Comment by hacksparrow** [comment]
Please send them to captain@hacksparrow.com, and specify your OS and ImageMagick version.

**Comment by gzurbach** [comment]
Thanks @hacksparrow for your quick answer. I did some debugging and figured out the source of the bug.
In the `info` method we run the `identify` command (see https://github.com/hacksparrow/node-easyimage/blob/master/easyimage.js#L36). 
The `%x` param returns `72 Undefined` because the unit of the density cannot be found (known ImageMagick bug: http://www.imagemagick.org/discourse-server/viewtopic.php?t=17451).

Since the `stdoud` is just split by spaces, `temp[6]` equals `Undefined`.
I'm using ImageMagick 6.7.7.10 on Debian Lenny. I'm gonna send the image by email in a minute.

**Comment by hacksparrow** [comment]
Just published easyimage to 2.0.0, see if it helped. If not, please share the output of `identify -format '%m %z %w %h %b %x %y %f' <YOUR FILE>`.

**Comment by gzurbach** [comment]
I just tried this morning and it still doesn't work. The value of stdout is:
`PNG 8 800 413 82617B 72 Undefined 72 Undefined image.png`
I think the fix is fairly simple, the line:
var temp = stdout.replace('PixelsPerInch', '').replace('PixelsPerCentimeter', '').split(' ');
Sould be:
var temp = stdout.replace('PixelsPerInch', '').replace('PixelsPerCentimeter', '').replace('Undefined', '').split(' ');
What do you think? I can test it and send it a PR if you want.

**Comment by hacksparrow** [comment]
This issue does not exist in the latest version of IM. Please send a pr
with the fix for the affected version.

**Comment by gzurbach** [comment]
Just FYI: `replace` only replaces the first occurence of the value in a string. Since we added `%y` we know have two occurences of `PixelsPerInch` or `PixelsPerCentimeter` and onyl one is removed. The v2.0 has a pretty bad bug!

**Comment by hacksparrow** [comment]
Thanks for noticing it and working on it. I am working with `ImageMagick 6.8.9-5`, so all looked good.

**Comment by gzurbach** [comment]
Do you know when you will publish the next release?

**Comment by hacksparrow** [comment]
In a while.

Isn't `stdout.replace(/PixelsPerInch/g, '')` with the `/g` flag not doing a global replace? You don't need to call it twice.

**Comment by hacksparrow** [comment]
Err, it's not the same thing. Please disregard the comment above.

**Comment by gzurbach** [comment]
Not sure I understand. The first call replaces `PixelsPerInch`, the second replaces `PixelsPerCentimeter`. 
The bug introduced in 2.0.0 is that stdoud looks like:
`JPEG 8 800 600 67762B 72 PixelsPerInch 72 PixelsPerInch image.jpg`

As you can see, `PixelsPerInch` is found twice but the `replace` was not global. So I fixed the bug by adding `/g`. You should release ASAP because 2.0.0 seems to have a pretty bad bug.

**Comment by hacksparrow** [comment]
Just published 2.0.1. Thanks for the contribution!

**Comment by gzurbach** [comment]
Awesome, you're welcome!
